### DISCUSSION TITLE
Distinguish "waiting" time from "busy" time

### DISCUSSION THREAD

**Discussion by alex-jitbit** [microsoft/perfview#1951] [Q&A] [answered]
I'm new to tracing so forgive me if the question sounds noob'ish

I'm profiling a .NET app trying to understand where the CPU usage comes from. Using `dotnet-trace collect -p PID` I record the trace file, and then view the trace on my local dev machine using PerfView.

In "thread time" view I see that most of the time is wasted in methods like `Monitor.Wait` or `LowLevelLifoSemaphore.WaitForSignal` which is obviously "blocked' and "waiting" code, that is not doing anything, just waiting for a signal to continue.

Is there a way to filter the view somehow, so it does not count "blocking" code like this?

**Comment by Xhanti** [comment, marked as answer]
Hi @alex-jitbit ,
If you haven't seen these: https://learn.microsoft.com/en-us/shows/perfview-tutorial/
, please take a look at them. They a bit old, but they tell about the philosophy of perfview and how to think about perf investigations.

For your case this might be interest: https://learn.microsoft.com/en-us/shows/perfview-tutorial/tutorial-14-investigating-wall-clock-responce-time-in-aspnet-scenarios

**Reply by alex-jitbit** [reply]
That video did help, thanks. I just needed to remove the "folding" that's there by default, and then start unfolding everything down form `CPU_TIME`. Thanks

**Reply by Xhanti** [reply]
My pleasure 👍🏽
